nixos/modules/services/web-apps/movim.nix

```diff
@@ -102,23 +102,27 @@ let
           in
           lib.concatStringsSep "\n" [
             (lib.optionalString brotli.enable ''
+              precompress_broti() {
+                file="$1"
+                ${lib.getExe brotli.package} --keep --quality=${builtins.toString brotli.compressionLevel} --output=$file.br $file
+              }
+
               echo -n "Precompressing static files with Brotli …"
-              find ${appDir}/public -type f ${findTextFileNames} \
-                | ${lib.getExe pkgs.parallel} ${lib.escapeShellArgs [
-                    "--will-cite"
-                    "-j $NIX_BUILD_CORES"
-                    "${lib.getExe brotli.package} --keep --quality=${builtins.toString brotli.compressionLevel} --output={}.br {}"
-                   ]}
+              find ${appDir}/public -type f ${findTextFileNames} | while read -r file; do
+                precompress_broti $file &
+              done
+              wait
               echo " done."
             '')
             (lib.optionalString gzip.enable ''
+              precompress_gzip() {
+                file="$1"
+                ${lib.getExe gzip.package} -c -${builtins.toString gzip.compressionLevel} $file > $file.gz
+              }
+
               echo -n "Precompressing static files with Gzip …"
-              find ${appDir}/public -type f ${findTextFileNames} \
-                | ${lib.getExe pkgs.parallel} ${lib.escapeShellArgs [
-                    "--will-cite"
-                    "-j $NIX_BUILD_CORES"
-                    "${lib.getExe gzip.package} -c -${builtins.toString gzip.compressionLevel} {} > {}.gz"
-                   ]}
+              find ${appDir}/public -type f ${findTextFileNames} | while read -r file; do
+                precompress_gzip $file &
```

Comment:
How many files do we have? This doesn't set any upper bound on how many processes are spawned. I am bit concerned about error handling here as well.